assignment_2_Script.js

```diff
@@ -0,0 +1,81 @@
+var dateCreation = (function iife() {
+  var currentModuleDate = null;
+  function instantiateDate() {
+    return new Date();
+  }
+  function setDate(date) {
+    if (date===undefined) {
+      currentModuleDate = instantiateDate();
+    }
+    else if (typeof date === 'number') {
+      currentModuleDate = new Date(date);
+    }
+    else if (date instanceof Date) {
+      currentModuleDate = new Date(date);
+    }
+  }
+  function getDate(requestedFormat) {
+    if (currentModuleDate === null) {
+      currentModuleDate = instantiateDate();
+    }
+    if (requestedFormat.format === 'formatted') {
```

Comment:
You want to check requestedFormat before you try to access a property on it. If you run your code and try:

`dateCreation.getDate()` you'll get an error, because `requestedFormat` is undefined.